lib/core/src/Cardano/Wallet/Api.hs

```diff
@@ -880,6 +886,31 @@ type DeleteSharedWallet = "shared-wallets"
     :> Capture "walletId" (ApiT WalletId)
     :> DeleteNoContent
 
+{-------------------------------------------------------------------------------
+                                  Shared Wallet Keys
+  See also: https://input-output-hk.github.io/cardano-wallet/api/#tag/Keys
+-------------------------------------------------------------------------------}
+
+type SharedWalletKeys =
+         GetSharedWalletKey
+    :<|> PostAccountKeyShared
+
+-- | https://input-output-hk.github.io/cardano-wallet/api/#operation/getSharedWalletKey
+type GetSharedWalletKey = "shared-wallets"
+    :> Capture "walletId" (ApiT WalletId)
+    :> "keys"
+    :> Capture "role" (ApiT Role)
+    :> Capture "index" (ApiT DerivationIndex)
+    :> Get '[JSON] ApiVerificationKeyShared
+
+-- | https://input-output-hk.github.io/cardano-wallet/api/#operation/postAccountKeyShared
+type PostAccountKeyShared = "shared-wallets"
+    :> Capture "walletId" (ApiT WalletId)
+    :> "keys"
+    :> Capture "index" (ApiT DerivationIndex)
+    :> ReqBody '[JSON] ApiPostAccountKeyData
+    :> PostAccepted '[JSON] ApiAccountKeyShared
```

Comment:
I don't think these are necessary; Surely the structure and outer interface is the same as for standard 1852' keys, right?

Reply:
IMHO it is small code bloating that serves well in improving code clarity. We have
```
1827     ApiVerificationKeyShelley: &ApiVerificationKeyShelley                                                                                                                                                     
1828       type: string                                                                                                                                                                                            
1829       format: bech32                                                                                                                                                                                          
1830       pattern: "^((addr_vk)|(stake_vk))1[0-9a-z]*$"                                                                                                                                                           
1831                                                                                                                                                                                                               
1832     ApiVerificationKeyShared: &ApiVerificationKeyShared                                                                                                                                                       
1833       type: string                                                                                                                                                                                            
1834       format: bech32                                                                                                                                                                                          
1835       pattern: "^((addr_shared_vk)|(stake_shared_vk))1[0-9a-z]*$"
```
And the same on Api.Types level, here ToJSON instances capturing those prefixes, and also enpoints consuming either `ApiVerificationKeyShelley` or `ApiVerificationKeyShared`. I think it is dead simply and ... that's good.